src/Chapter2.hs

```diff
@@ -841,7 +854,12 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: (Num a) => Int -> [a] -> [a]
+rotate n ls
+  | n < 0 = []
+  | otherwise = take listLen . drop n . cycle $ ls
+  where
+    listLen = length ls
```

Comment:
Your implementation is correct. Though, you can optimize it even further. Note, that if `n` is a big number, `drop` will do a lot of work, if the list size is much smaller than the number `n`. It looks like it possible to optimize this function a bit :slightly_smiling_face: